### PR TITLE
HTTP/2 code sharing shell sync-script

### PR DESCRIPTION
**Pull request by gfoidl**
See https://github.com/dotnet/corefx/pull/42070#discussion_r343350111

/cc: @Tratcher

### CODE REVIEW

**Comment by Tratcher** [comment]
Hmm, this should have automatically required a review from dotnet/http2 as a codeowner. Maybe we need trailing slashes on those paths?
https://github.com/dotnet/corefx/blob/46edc58620f29557e23fea29ed8392a0f2c9e31c/.github/CODEOWNERS#L4-L5

**Comment by Tratcher** [comment]
@gfoidl Thanks. The best way to test these is to use them to copy themselves over to aspnet/aspnetcore. Can you try that and submit the matching PR there?

File: src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh
Comment (am11) [inline src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh]: * in OS like FreeBSD, bash is not available in this location in default settings, therefore most (if not all) scripts in dotnet repos use `#!/usr/bin/env bash`, i.e. a more portable shebang.
* in this script, there is no particular need for bash (mostly used for string replacement, arrays etc. facilities); so we can  switch to even more portable POSIX shell syntax: `#!/usr/bin/env sh` and replace `if [[ .. ]]` with `if [ .. ]`.

Comment (gfoidl) [reply src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh]: Thanks 👍

**Comment by gfoidl** [comment]
> Can you try that and submit the matching PR there?

https://github.com/aspnet/AspNetCore/pull/16885

File: src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh
Comment (eiriktsarpalis) [inline src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh]:
```suggestion
    echo The 'ASPNETCORE_REPO' environment variable or command line parameter is not set, aborting.
```

File: src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh
Comment (eiriktsarpalis) [inline src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh]: These relative paths look scary :-) 

Consider adding a ``cd `dirname $0` `` statement at the top of the script to ensure that these always evaluate relative to the script as opposed to the caller's working directory.

File: src/Common/src/System/Net/Http/Http2/CopyToCoreFx.sh
Comment (eiriktsarpalis) [inline src/Common/src/System/Net/Http/Http2/CopyToCoreFx.sh]:
```suggestion
    echo The 'COREFX_REPO' environment variable or command line parameter is not set, aborting.
```

File: src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh
Comment (eiriktsarpalis) [inline src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh]: We generally assume bash in most of our scripts in this repo.

Comment (gfoidl) [reply src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh]: I changed that based on https://github.com/dotnet/corefx/pull/42425#discussion_r343498168

File: src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh
Comment (gfoidl) [reply src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh]: Thanks. 

Copied that from https://github.com/dotnet/corefx/blob/70bdacc3e9fcb50acce9e4a3045e35452b5a3410/src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.cmd#L7
Will fix this too.

File: src/Common/src/System/Net/Http/Http2/CopyToCoreFx.sh
Comment (gfoidl) [reply src/Common/src/System/Net/Http/Http2/CopyToCoreFx.sh]: Same here. Thanks.

File: src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh
Comment (gfoidl) [reply src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh]: Good idea 👍

File: src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh
Comment (eiriktsarpalis) [reply src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh]: Scripts in corefx use bash. For the sake of consistency I'd argue it should be calling bash here as well (even if it doesn't -currently- use bash-specific constructs).

Comment (gfoidl) [reply src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh]: > Scripts in corefx use bash

Yep, that's true -> https://github.com/dotnet/corefx/search?p=5&q=bash&unscoped_q=bash
So I'll change this to `#!/usr/bin/env bash` as everywhere else.

File: src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh
Comment (am11) [reply src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh]: Bash is a POSIX shell complaint, so it is a superset of POSIX shell. `if [ .. ]` works in both, `if [[ .. ]]` works only in Bash. Unless there is a need need for `[[]]` (like pattern match or array access), `[]` works fine. Scripts in .NET core org use both `[]` and `[[]]` (latter one, when really needed).

Comment (gfoidl) [reply src/Common/src/System/Net/Http/Http2/CopyToAspNetCore.sh]: While true, I don't see the point here for this script, which will be used in a relatively advanced scenario unlike `build.sh`, etc. It's for inner-loop development of HTTP2, so this should just be fine, as it does it's job of syncing / mirroring between the two repos.

**Review by Tratcher — APPROVED** [review]

**Review by halter73 — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks @gfoidl

**Pull request merged**
